### PR TITLE
Feature/vt bp review

### PR DESCRIPTION
**Pull request by cwickert**
I finished the review of the Virtualization Best practices. I covered a lot, but still don't think it's quite finished.

Open issues:
* inconsistent use of emphasis vs. literal
* inconsistent use of prompts and screens

### CODE REVIEW

**Comment by cwickert** [comment]
[[DOCU] Virtualization Best Practices Guide](https://trello.com/c/bcm9xL2X/195-docu-virtualization-best-practices-guide)

File: xml/vt_best_practices.xml [hunk 1]
```diff
@@ -2805,7 +2800,8 @@ virtio-net-pci.any_layout=on/off
     a better entropy in the &vmguest;.
    </para>
    <para>
-    To use the <filename>hwrng</filename>, add the device in the XML
+    To use the <filename>hwrng</filename>, add an <literal>RNG</literal> device
```
Comment (fsundermeyer) [inline xml/vt_best_practices.xml]: Please explain RNG at the beginning of this section:

Virtio RNG (random number generator) is a paravirtualized device ...

and then replace

<filename>hwrng</filename> with "Virtio RNG"

File: xml/vt_best_practices.xml [hunk 2]
```diff
@@ -2950,6 +2946,8 @@ virtio_rng.0</screen>
 
   <sect2 xml:id="vt.best.perf.mtype">
    <title>Updating the Guest Machine Type</title>
+   <!-- cwickert 2017-01-26: The sentences in the next two paragaphs are quite
+    long and hard to read/understand. Try to make them shorter? -->
```
Comment (fsundermeyer) [inline xml/vt_best_practices.xml]: &qemu; machine types define details of the architecture that are particularly relevant during migration and session management. The list of machine types is constantly maintained&mdash;the addition of new features and fixes results in new machine types.
Old machine types remain in the list for compatibility reasons when new versions are released. However, to take advantage of improvements, it is recommended to always migrate to the latest machine type when upgrading.

File: xml/vt_best_practices.xml [hunk 3]
```diff
@@ -2987,7 +2985,7 @@ virtio_rng.0</screen>
    <para>
     The ability to change a &vmguest;'s state heavily depends on the
     operating system. It is very important to test this feature before any
-    use of your &vmguest; in production. For example most Linux operating
+    use of your &vmguest; in production. For example, most Linux operating
```
Comment (fsundermeyer) [inline xml/vt_best_practices.xml]: ...befor using your &vmguest;s...

File: xml/vt_best_practices.xml [hunk 4]
```diff
@@ -3052,23 +3049,22 @@ error: Failed to start domain sles12
...
 2015-08-12T11:21:14.221634Z qemu-system-x86_64: failed to initialize spice server</screen>
    <para>
-    To change this you can change the default
-    <emphasis>spice_listen</emphasis> value in
-    <filename>/etc/libvirt/qemu.conf</filename> using the local ipv6 address
-    <emphasis>::1</emphasis>. The spice server listening address can also be
-    changed on a per &vmguest; basis, use <command>virsh edit</command> to
-    add the listen XML attribute to the <emphasis>graphics
+    To fix this, you can change the default <emphasis>spice_listen</emphasis>
+    value in <filename>/etc/libvirt/qemu.conf</filename> using the local ipv6
```
Comment (fsundermeyer) [inline xml/vt_best_practices.xml]: &lt;literal>spice_listen&lt;/literal>

File: xml/vt_best_practices.xml [hunk 5]
```diff
@@ -3052,23 +3049,22 @@ error: Failed to start domain sles12
...
+    address <emphasis>::1</emphasis>. The spice server listening address can
```
Comment (fsundermeyer) [inline xml/vt_best_practices.xml]: &lt;literal>1&lt;/literal>

File: xml/vt_best_practices.xml [hunk 6]
```diff
@@ -3052,23 +3049,22 @@ error: Failed to start domain sles12
...
 
-  <sect2>
+  <sect2 xml:id="vt.best.xml_to_qemu">
    <title>XML to QEMU command line</title>
    <para>
     Sometimes it could be useful to get the QEMU command line to launch the
-    &vmguest; from the XML file. Use <command>virsh</command> with
-    <option>domxml</option>.
+    &vmguest; from the XML file. Use <command>virsh</command> with the
+    <option>domxml</option> option.
```
Comment (fsundermeyer) [inline xml/vt_best_practices.xml]: I'd rather delete this sentence

File: xml/vt_best_practices.xml [hunk 7]
```diff
@@ -3828,7 +3824,7 @@ VGA trouble debug
      <para>
       Run the sandbox, using <command>virt-sandbox</command>. This command
       has many interesting options, read its man page to discover them all.
-      It can either be running as root or as an unprivileged user.
+      The command can be run as root or as an unprivileged user.
```
Comment (fsundermeyer) [inline xml/vt_best_practices.xml]: ...&rootuser;...

File: xml/vt_best_practices.xml [hunk 8]
```diff
@@ -3879,14 +3876,14 @@ VGA trouble debug
...
-<sect1 xml:id="vt.best.Xen_pv_fv">
+<sect1 xml:id="vt.best.xen_pv_fv">
   <title>&xen;: Moving from PV to FV</title>
   <para>
    This chapter explains how to convert a &xen; para-virtual machine into a
    &xen; fully virtualized machine.
   </para>
   <para>
-   First you need to change to the <emphasis>-default</emphasis> kernel, if not
+   First you need to change to the <emphasis>-default</emphasis> kernel; if not
```
Comment (fsundermeyer) [inline xml/vt_best_practices.xml]: First you need to change to the default Kernel (package <package>kernel-default</papckage>); if not
￼already installed, you need to install it while in PV mode.

_Note: No "must", it should rather be "need to" (according to the Styleguide)._

File: xml/vt_best_practices.xml (hunk 5 above)
Comment (cwickert) [inline xml/vt_best_practices.xml]: <systemitem class="ipaddress">::1</systemitem>

Comment (cwickert) [inline xml/vt_best_practices.xml]: &lt;systemitem class="ipaddress"&gt;::1&lt;/systemitem&gt;

File: xml/vt_best_practices.xml (hunk 1 above)
Comment (cwickert) [inline xml/vt_best_practices.xml]: Done

File: xml/vt_best_practices.xml (hunk 2 above)
Comment (cwickert) [inline xml/vt_best_practices.xml]: I've use a yer shorter combination of the original and your suggestion. Thanks!

File: xml/vt_best_practices.xml (hunk 3 above)
Comment (cwickert) [inline xml/vt_best_practices.xml]: Fixed

File: xml/vt_best_practices.xml (hunk 4 above)
Comment (cwickert) [inline xml/vt_best_practices.xml]: Fixed

File: xml/vt_best_practices.xml (hunk 6 above)
Comment (cwickert) [inline xml/vt_best_practices.xml]: Done.

File: xml/vt_best_practices.xml (hunk 7 above)
Comment (cwickert) [inline xml/vt_best_practices.xml]: Fixed.

File: xml/vt_best_practices.xml (hunk 8 above)
Comment (cwickert) [inline xml/vt_best_practices.xml]: Fixed.

**Comment by cwickert** [comment]
Thanks for the review! IFAICS I addressed all of your points (and many more). I've also spell-, link, and -stylechecked everything.

The only remaining warnings/erros from stylechecker are

* some long sentences
* missing introductory paragraphs or abstracts on some sections. IHMO this is ok for a best practice.
* > Deliverable article.vt.best.practices does not contain a date.
  >       
  >      Add a date element, use <?dbtimestamp?> inside.
  Do we add dates to Best Practices?

**Comment by fsundermeyer** [comment]
Yes, I think we should add a date - it will tell people when the guide has been updated.

**Comment by fsundermeyer** [comment]
Everything is fine now, thanks a lot!

**Pull request merged**